IngestionGrain/Business.cs

```diff
@@ -0,0 +1,82 @@
+﻿using CommunAxiom.Commons.Client.Contracts.Datasource;
+using CommunAxiom.Commons.Client.Contracts.Grains.Storage;
+using CommunAxiom.Commons.Client.Contracts.Ingestion;
+using CommunAxiom.Commons.Ingestion;
+using Newtonsoft.Json.Linq;
+using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CommunAxiom.Commons.Client.Grains.IngestionGrain
+{
+    public class Business
+    {
+        private Repo _repo;
+
+        private readonly Importer _importer;
+        private readonly IGrainFactory _grainFactory;
+        private readonly string _grainKey;
+
+        public Business(Importer importer, IGrainFactory grainFactory, string grainKey)
+        {
+            _importer = importer;
+            _grainFactory = grainFactory;
+            _grainKey = grainKey;
+        }
+
+        public void Init(IPersistentState<IngestionHistory> history)
+        {
+            _repo = new Repo(history);
+        }
+
+        public Task<IngestionHistory> GetHistory()
+        {
+            return _repo.FetchHistory();
+        }
+
+        public async Task Run()
+        {
+            try
+            {
+                var dataSource = _grainFactory.GetGrain<IDatasource>(_grainKey);
```

Comment:
You need to keep track of the ingestion progress in the business object to that you know if an ingestion is in progress or not. You shouldn't be able to call run on the same datasource twice simultaneously (can't have two identical ingestions at the same time). The method run should return an error is an ingestion is in progress.